src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+        /// <returns>Array of pixel temperatures [column, row]</returns>
+        public Temperature[,] GetThermalImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new Temperature[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = Amg88xxUtils.ConvertToTemperature(tl, th);
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the current raw thermal image from the sensor.
+        /// </summary>
+        /// <returns>Thermal image in two's complement representation</returns>
+        public int[,] GetThermalRawImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new int[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = th << 8 | tl;
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the temperature reading from the internal thermistor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetSensorTemperature()
+        {
+            byte tthl = GetRegister(Register.TTHL);
+            byte tthh = GetRegister(Register.TTHH);
+
+            int reading = (tthh & 0x7) << 8 | tthl;
+            reading = tthh >> 3 == 0 ? reading : -reading;
+
+            // The LSB is equivalent to 0.0625℃.
+            return Temperature.FromDegreesCelsius(reading * ThermistorTemperatureResolution);
+        }
+
+        #endregion
+
+        #region Status
+
+        /// <summary>
+        /// Gets the temperature overflow flag from the status register
+        /// </summary>
+        /// <returns>Temperature overflow flag</returns>
+        public bool HasTemperatureOverflow()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.OVF_IRS);
+        }
+
+        /// <summary>
+        /// Clears the temperature overflow flag in the status register
+        /// </summary>
+        public void ClearTemperatureOverflow()
```

Comment:
consider collapsing this and previous method into one and doing the clearing automatically, start with writing a sample showing the usage and based on that sample figure out if it makes more sense to have one or two methods.

Also if you keep two methods make sure to describe this scenario more. When should user call this? What happens if they don't (i.e. will reading temperature not work at all?)